torch/csrc/distributed/c10d/init.cpp

```diff
@@ -701,10 +701,10 @@ Example::
               std::chrono::milliseconds>(),
           py::arg("host_name"),
           py::arg("port"),
-          py::arg("world_size"),
+          py::arg("world_size") = -1,
```

Comment:
Since this is through pybind, I was not able to create an optional None argument without the extra complexity of defining a separate class for this argument. Pybind supports using [std::optional](https://pybind11.readthedocs.io/en/stable/advanced/functions.html#allow-prohibiting-none-arguments) but we are on C++ 14 which does not have std::optional so I decided to leave it as is. However, I did change the C++ constructor and I will add the waitWorkers param / getPort method in a separate PR.

Reply:
Although not "Pythonic", fair enough. As a side note; pybind11's support for standard container types can be extended via a `type_caster<>`. If you have a type that is API compatible with an STL type (e.g. `c10::optional`), you can easily support it in your Python interop. See [here](https://docs.google.com/document/d/15Caw9eGSE949xljF59YZb2FPR2Z762Pc9GVZU4aVFh4) for more info.